tests/tests_geomstats/test_positive_lower_triangular_matrices.py

```diff
@@ -0,0 +1,291 @@
+
+        mats = gs.array([[1.0, -1.0], [-1.0, 3.0]])
+        result = self.space.belongs(mats)
+        expected = False
+        self.assertAllClose(result, expected)
+
+        mats = gs.array([[-1.0, 0.0], [-1.0, 3.0]])
+        result = self.space.belongs(mats)
+        expected = False
+        self.assertAllClose(result, expected)
+
+        mats = gs.eye(3)
+        result = self.space.belongs(mats)
+        expected = False
+        self.assertAllClose(result, expected)
+
+    def test_belongs_vectorization(self):
+        """Test of belongs method."""
+        mats_2dim = gs.array(
+            [
+                [[1.0, 0], [0, 1.0]],
+                [[1.0, 2.0], [2.0, 1.0]],
+                [[-1.0, 0.0], [1.0, 1.0]],
+                [[0.0, 0.0], [1.0, 1.0]],
+            ]
+        )
+
+        mats_3dim = gs.array(
+            [
+                [[1.0, 0.0, 1.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                [[0.0, 0.0, 0.0], [2.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                [[1.0, 0.0, 0.0], [2.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                [[-1.0, 0.0, 0.0], [2.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+            ]
+        )
+
+        result = self.space.belongs(mats_2dim)
+        expected = gs.array([True, False, False, False])
+        self.assertAllClose(result, expected)
+
+        result = PositiveLowerTriangularMatrices(3).belongs(mats_3dim)
+        expected = gs.array([False, False, True, False])
+        self.assertAllClose(result, expected)
+
+    def test_random_point_and_belongs(self):
+        """Test of random_point and belongs methods."""
+        point = self.space.random_point()
+        result = self.space.belongs(point)
+        expected = True
+        self.assertAllClose(result, expected)
+
+    def test_random_point_and_belongs_vectorization(self):
+        """Test of random_point and belongs methods."""
+        points = self.space.random_point(4)
+        result = self.space.belongs(points)
+        expected = gs.array([True] * 4)
+        self.assertAllClose(result, expected)
+
+    def test_gram(self):
+        """Test gram method for single point"""
+        L = gs.array([[1.0, 0.0], [2.0, 1.0]])
+        gram_expected = gs.array([[1.0, 2.0], [2.0, 5.0]])
+        gram_result = self.space.gram(L)
+        self.assertAllClose(gram_expected, gram_result)
+
+    def test_gram_vectorization(self):
+        """Test gram method for batch of points"""
+        point = self.space.random_point(n_samples=5)
+        gram = self.space.gram(point)
+
+        gram_result = gram
+        gram_expected = gs.matmul(point, Matrices.transpose(point))
+        self.assertAllClose(gram_expected, gram_result)
+
+        belongs_result = gs.all(SPDMatrices(self.n).belongs(gram))
+        belongs_expected = True
+        self.assertAllClose(belongs_expected, belongs_result)
+
+    def test_differential_gram(self):
+        """Test differential of gram for single point"""
+        L = gs.array([[1.0, 0.0], [2.0, 1.0]])
```

Comment:
Avoid using uppercase letters for variables, these are reserved for classes, see PEP8:
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Also, let's avoid using single letters as variable names: it makes "find-and-replace" very complicated during refactoring. Thanks!